File: resources/algorithmic-toolbox/week2_algorithmic_warmup/1_fibonacci_number/fibonacci.js
// by Alexander Nikolskiy

const readline = require('readline');

async function main (){
    const rl = readline.createInterface({
        input: process.stdin,
        terminal: false
    });
    
    process.stdin.setEncoding('utf8');

    for await (const line of rl) {
        console.log(fib(parseInt(line, 10)));
        process.exit();
    }
}

function fib(n) {
    const f_list = [BigInt(0), BigInt(1)]

    for (let i = 2; i <= n; i++) {
        f_list.push(f_list[i - 1] + f_list[i - 2]);
    }

    return f_list[n].toString();
}

module.exports = fib;

if (!module.parent) {
    main();
}

File: resources/algorithmic-toolbox/week2_algorithmic_warmup/1_fibonacci_number/fibonacci.py
# Uses python3
def calc_fib(n):
    if (n <= 1):
        return n

    f_list = [0, 1]
    
    for i in range(2, n + 1):
        f_list.append(f_list[i - 1] + f_list[i - 2])

    return f_list[n]

n = int(input())
print(calc_fib(n))
